backend/src/main/java/gov/cdc/usds/simplereport/logging/QueryLoggingInstrumentation.java

```diff
@@ -0,0 +1,58 @@
+package gov.cdc.usds.simplereport.logging;
+
+import graphql.ExecutionResult;
+import graphql.execution.ExecutionId;
+import graphql.execution.instrumentation.InstrumentationContext;
+import graphql.execution.instrumentation.SimpleInstrumentation;
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters;
+import graphql.execution.instrumentation.parameters.InstrumentationValidationParameters;
+import graphql.language.Field;
+import graphql.language.OperationDefinition;
+import graphql.validation.ValidationError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Created by nickrobison on 11/27/20
+ */
+public class QueryLoggingInstrumentation extends SimpleInstrumentation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(QueryLoggingInstrumentation.class);
+
+    @Override
+    public InstrumentationContext<List<ValidationError>> beginValidation(InstrumentationValidationParameters parameters) {
+
+        // Descend through the GraphQL query and pull out the field names and variables from the operation definitions
+        final Set<String> fieldSet = parameters
+                .getDocument()
+                .getDefinitions()
+                .stream()
+                .filter(definition -> definition instanceof OperationDefinition)
```

Comment:
I would have thought that `isAssignableFrom` would have been a good method for this so you don't need a lambda, but I would have been wrong.